tuf/store/interfaces.go

```diff
@@ -12,7 +17,8 @@ type MetadataStore interface {
 
 // PublicKeyStore must be implemented by a key service
 type PublicKeyStore interface {
-	GetKey(role string) ([]byte, error)
+	GetKey(role string) (data.PublicKey, error)
+	RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error)
```

Comment:
The stores are meant to be reasonably low level and it feels like they're losing cohesion when they start referencing `PublicKey` and `CryptoService`.